### PR TITLE
#137005155 make endpoints restful api

### PR DESCRIPTION
**Pull request by vonvick**
#### What does this PR do?
This PR makes the API endpoints RESTful
#### Description of Task to be completed?
Extend Meteor DDP methods and provide restful API.
#### How should this be manually tested?
it can be tested by using POSTMAN to test the different API endpoints
open POSTMAN and enter <your domain>/api/<collectionName>
e.g GET http://localhost:3000/api/Products
#### Any background context you want to provide?
No background context
#### What are the relevant pivotal tracker stories?
#137005155 Make the API endpoints RESTful
#### Screenshots (if appropriate)
api-endpoint

#### Questions:
No Questions

### CODE REVIEW

File: server/startup/restful-endpoints.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Accounts, Cart, Orders, Products, Shops } from "/lib/collections";
+
+export default () => {
+  // Global API configuration
+  const Api = new Restivus({
```
[the remark this answers is not in the document]
Comment (vonvick) [reply server/startup/restful-endpoints.js:5]: fixed

File: server/startup/restful-endpoints.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { Accounts, Cart, Orders, Products, Shops } from "/lib/collections";
+
+export default () => {
```
Comment (vonvick) [inline server/startup/restful-endpoints.js:3]: fixed

File: server/startup/restful-endpoints.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+  // Global API configuration
+  const Api = new Restivus({
```
[the remark this answers is not in the document]
Comment (vonvick) [reply server/startup/restful-endpoints.js:5]: fixed

File: server/startup/restful-endpoints.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          action() {
+            const isUpdated = collection.update(this.urlParams.id, {
+              $set: this.bodyParams
+            });
+            if (isUpdated) {
+              const record = collection.findOne(this.urlParams.id);
+              return {
+                status: "success",
+                data: record
+              }
```
[the remark this answers is not in the document]
Comment (vonvick) [reply server/startup/restful-endpoints.js]: fixed

File: server/startup/restful-endpoints.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+            }
+            return {
+              statusCode: 404,
```
[the remark this answers is not in the document]
Comment (vonvick) [reply server/startup/restful-endpoints.js]: fixed

File: server/startup/restful-endpoints.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+              return {
+                status: "success",
+                data: record
+              }
+            }
+            return {
+              statusCode: 404, 
+              message: "Item not found",
+              body: this.bodyParams
+            }
```
[the remark this answers is not in the document]
Comment (vonvick) [reply server/startup/restful-endpoints.js]: fixed

**Comment by olawalejarvis** [comment]
The code looks okay to me

**Review by andela-aolaniran — COMMENTED** [review]
Looks good.

**Comment by vonvick** [comment]
Could not find any error flagged by hound

**Review by victor-enogwe — APPROVED** [review]
hound flagging .gitignore for no reason, will merge moving forward looks pretty solid and good

**Pull request merged**